polyfill/lib/ecmascript.mjs

```diff
@@ -1570,6 +1573,30 @@ export const ES = ObjectAssign({}, ES2019, {
     );
   },
 
+  CreateTemporalDate: (isoYear, isoMonth, isoDay, calendar, object) => {
+    ES.RejectDate(isoYear, isoMonth, isoDay);
+    ES.RejectDateRange(isoYear, isoMonth, isoDay);
+    if (!calendar || typeof calendar !== 'object') throw new RangeError('invalid calendar');
+    if (object === undefined) object = new GetIntrinsic('%Temporal.Date%');
+    CreateSlots(object);
+    SetSlot(object, ISO_YEAR, isoYear);
+    SetSlot(object, ISO_MONTH, isoMonth);
+    SetSlot(object, ISO_DAY, isoDay);
+    SetSlot(object, CALENDAR, calendar);
+    return object;
+  },
+
+  CompareTemporalDate: (one, two) => {
+    for (const slot of [ISO_YEAR, ISO_MONTH, ISO_DAY]) {
+      const val1 = GetSlot(one, slot);
+      const val2 = GetSlot(two, slot);
+      if (val1 !== val2) return ES.ComparisonResult(val1 - val2);
+    }
+    const cal1 = GetSlot(one, CALENDAR).id;
+    const cal2 = GetSlot(two, CALENDAR).id;
```

Comment:
```suggestion
    const cal1 = ES.CalendarToString(GetSlot(one, CALENDAR));
    const cal2 = ES.CalendarToString(GetSlot(two, CALENDAR));
```

with 
```js
  CalendarToString: (calendar) => {
    let toString = calendar.toString;
    if (toString === undefined) {
      toString = GetIntrinsic('%Temporal.Calendar.prototype.toString%');
    }
    return ES.ToString(ES.Call(toString, calendar));
  },
```

This guarantees that the `<`/`>` operators aren't author-observable.

Reply:
Done.